contracts/SportMarkets/Parlay/ParlayVerifier.sol

```diff
@@ -210,43 +208,86 @@ contract ParlayVerifier {
 
     function _checkRisk(
         address[] memory _sportMarkets,
-        uint[] memory _positions,
         uint _sUSDInRisk,
         address _parlayAMM
     ) internal view returns (bool riskFree) {
         uint riskCombination;
         if (_sportMarkets.length == 2) {
-            riskCombination = IParlayMarketsAMM(_parlayAMM).riskPerCombination(
+            riskCombination = IParlayMarketsAMM(_parlayAMM).riskPerGameCombination(
                 _sportMarkets[0],
-                _positions[0],
                 _sportMarkets[1],
-                _positions[1],
                 address(0),
-                0,
                 address(0),
-                0
+                address(0),
+                address(0),
+                address(0),
+                address(0)
             );
         } else if (_sportMarkets.length == 3) {
-            riskCombination = IParlayMarketsAMM(_parlayAMM).riskPerCombination(
+            riskCombination = IParlayMarketsAMM(_parlayAMM).riskPerGameCombination(
                 _sportMarkets[0],
-                _positions[0],
                 _sportMarkets[1],
-                _positions[1],
                 _sportMarkets[2],
-                _positions[2],
                 address(0),
-                0
+                address(0),
+                address(0),
+                address(0),
+                address(0)
             );
         } else if (_sportMarkets.length == 4) {
-            riskCombination = IParlayMarketsAMM(_parlayAMM).riskPerCombination(
+            riskCombination = IParlayMarketsAMM(_parlayAMM).riskPerGameCombination(
```

Comment:
is there really no better way to write this than 8 if/else statements?

Reply:
yea, there can be a for loop, updating values to a pre-defined array of 8 address(0), but not sure if that's better computationally and for the stack